# Homework 01: Convexity and introductory linear programming

**ISyE 524 — Fall 2026**  
**Due: September 14, 2026.** Consult Gradescope for the submission time.  
**Coverage:** Material through Thursday, September 10.

**Name:** Enter your name here.

This homework contains three problems. For Problem 1, justify your mathematical conclusions. For Problems 2 and 3, write your mathematical formulation before implementing it in Julia/JuMP. Define variables and units, state constraints and the objective, and interpret the resulting solution.

Work in your copy under `student-work/hw01/`. Use the course Julia environment and the Julia 1.12 kernel. Complete all requested Markdown and code cells, run the notebook from top to bottom, and save it with the results visible. Export the completed notebook to PDF using the course export task, review the PDF, and submit it to Gradescope with pages matched to the questions. A clear handwritten graph or formulation embedded as an image is acceptable.

End your submission with the required **Collaboration and LLM-use statement**. LLMs may help you learn, debug your own code, or critique your work; they should not replace constructing your formulation, implementation, and written solution. You are responsible for understanding and checking everything you submit.


## Before you begin HW01: update and create your working copy

1. **Open your course repository folder in VS Code**—the entire `isye524-students-julia` folder you used for HW0.

2. **Update the course files and Julia environment.** Open the Command Palette (**Ctrl+Shift+P** on Windows/Linux, **Cmd+Shift+P** on macOS), select **Tasks: Run Task**, and choose:
   ```text
   ISyE 524: Update course repository
   ```
   Wait for both the download and environment refresh to finish successfully.

3. **Create your personal HW01 copy.** From **Tasks: Run Task**, choose:
   ```text
   ISyE 524: Start an assignment from its template
   ```
   Enter **`hw01`** when prompted.

4. **Open `student-work/hw01/hw01.ipynb`.** Select the Julia 1.12 kernel and complete your work there. Keep the published template under `assignments/hw01/` unchanged.

**Already started?** Open your existing notebook under `student-work/hw01/`. The copy task refuses to overwrite existing work. Later course updates do not automatically update your personal assignment copy.

**If an update fails**, read the task’s error message and ask course staff for help before discarding any files. Back up your `student-work/` folder separately; Git does not track your answers.


## How to submit written answers and graphs

For **every written part**, you may either type your answer in Markdown using LaTeX mathematics, or insert a clear photograph/scan of handwritten work. Include your answer and its reasoning together, and label the subparts. You do not need to type a second copy of handwritten work. Julia/JuMP implementations should still be entered in code cells.

**Typed mathematics:** Double-click the response Markdown cell to edit it. Use `$x_1 \geq 0$` for inline mathematics or `$$x_1+x_2\leq5$$` for a displayed equation. Run the Markdown cell to render it.

**Photographs/scans:** Save a legible, cropped PNG or JPEG (convert HEIC or PDF scans to images first). In VS Code, edit the response Markdown cell, drag the image into it, and choose **Insert Image as Attachment**. This embeds the image inside the notebook so it travels with your `.ipynb` file.

**Keeping image files instead:** After starting `hw01` with the course assignment-copy task, open `student-work/hw01/hw01.ipynb`. Put your images in `student-work/hw01/images/`, for example `problem1.png`, and insert this in the response Markdown cell:

```markdown
![Problem 1: handwritten answer and reasoning](images/problem1.png)
```

The path is relative to your notebook. Keep the notebook and its `images/` folder together when copying or backing up your work. Put personal work in `student-work/hw01/`, not the tracked template in `assignments/hw01/`. `student-work/` is ignored by Git: pulling course updates is not a backup of your answers or images. Back up your working folder separately.

Before submitting, save the notebook, export it with the course PDF task, and check that every equation and image is present, readable, and in question order.

## Problem 1 — Convexity and what it tells us

Use definitions, convexity rules, or explicit counterexamples to justify your answers. A classification without a reason is incomplete. You do not need calculus, eigenvalues, or a solver. A sketch may help explain a counterexample, but a numerical solver result is not a proof of convexity.

### 1(a)–1(c): Functions

Classify each function as convex, concave, both, or neither on its stated domain.

**(a)** $f(x_1,x_2)=(x_1+x_2-1)^2$ on $\mathbb{R}^2$.

**(b)** $f(x_1,x_2)=|x_1|-|x_2|$ on $\mathbb{R}^2$.

**(c)** $f(x)=x^3$: first on $[0,\infty)$, then on $\mathbb{R}$. You may use the fact from lecture that $x^3$ is convex on $[0,\infty)$. Explain why the domain matters.


**Answer and reasoning for 1(a)–1(c):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.


**(a)** $f(x_1,x_2)=(x_1^2+x_2^2+2x_1x_2-x_1-x_2+1)$ on $\mathbb{R}^2$. This is convex because when it is multiplied out each of its cononents is either linear, or a quadratic with a non negitive leading constant which are both convex and a sum of convex functions is convex.


**(b)** $f(x_1,x_2)=|x_1|-|x_2|$ on $\mathbb{R}^2$. This function is neither convex or concave because it is the difference of two convex functions(or the sum of a convex and a concave function) which results in a funciton that is not convex or concave.

**(c)** $f(x)=x^3$: first on $[0,\infty)$, then on $\mathbb{R}$. This function is convex in the domain $[0,\infty]$ but not in $\mathbb{R}$ because power functions when the power is greater than or equal to 1 are only convex when $x\leq0$. 



### 1(d)–1(f): Sets

Determine whether each set is convex. For a nonconvex set, provide two points in the set whose midpoint is outside it.

**(d)** $D=\{(x_1,x_2)\in\mathbb{R}^2:x_1^2+x_2^2\leq 1\}$.

**(e)** $S=\{(x_1,x_2)\in\mathbb{R}^2:x_1^2+x_2^2=1\}$.

**(f)** $P=\{x\in\mathbb{R}^n:Ax\leq b\}$, where $A\in\mathbb{R}^{m\times n}$ and $b\in\mathbb{R}^m$. Justify your conclusion directly using a convex combination. Does it still apply if $P$ is empty?


**Answer and reasoning for 1(d)–1(f):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

**(d)** $D=\{(x_1,x_2)\in\mathbb{R}^2:x_1^2+x_2^2\leq 1\}$. Since the function $f(x_1,x_2)=(x_1+x_2-1)^2$ on $\mathbb{R}^2$ is convex the inequality which makes a set unclude all points less than a convex function is a convex set.  This is called a sublevel set. 

**(e)** $S=\{(x_1,x_2)\in\mathbb{R}^2:x_1^2+x_2^2=1\}$. This is not a convex set becasue it is produces a set which is comprized of all points landing on a circle with radius 1. If you select any two points on the circle the points within the line dont fall within the set.  Example if you select point $(1,1)$ and point $(-1,-1)$ which are within the set its midpoint is $(0,0)$ which is not in the set.

**(f)** $P=\{x\in\mathbb{R}^n:Ax\leq b\}$, where $A\in\mathbb{R}^{m\times n}$ and $b\in\mathbb{R}^m$. Since $Ax$ is a convex combination of linear functions, it is both convex and concave. Since it is less than $b$ it is a convex set as it is a sublevel set. $w=\alpha *x_1 + (1-\alpha)*x_2$ falls within $Ax\leq b$ thus it is a convex combination and therfore a convex set.  If P were to be empty the set would remain convex because $w=\alpha *x_1 + (1-\alpha)*x_2$ falls within $Ax\leq b$ when all x's are 0 still holds true.



### 1(g)–1(h): Optimization problems and guarantees

For each problem, decide whether it is a convex optimization problem. Address both the objective (including whether it is minimized or maximized) and the feasible region.

**(g)**
$$
\begin{aligned}
\max_{x_1,x_2\in\mathbb{R}}\quad &-|3x_1-2x_2|+\sqrt{4x_1}\\
\text{subject to}\quad &x_1\geq0,\\
&-14x_1+x_2=7.
\end{aligned}
$$

**(h)**
$$
\begin{aligned}
\min_{x_1,x_2\in\mathbb{R}}\quad &x_1^2+3x_2\\
\text{subject to}\quad &x_1^2+x_2^2\leq1.
\end{aligned}
$$
Also classify the problem obtained by changing **min** to **max** in (h).

For (g), (h), and the maximization version of (h), state whether convexity guarantees that every local optimum is global. Explain why “not a convex optimization problem” does **not** by itself prove that a particular instance will be hard to solve. Does convexity alone guarantee that an optimal solution exists and is unique?


**Answer and reasoning for 1(g)–1(h):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

**(g)** The objective function in this equation is maximizing a concave function which makes it a convex optimization probmem as long as the feasable region is also convex.  The feasable region is convex becaue while $x_1 \geq 0$ is convex the second constraint is convex as well becasue it is an affine equality. Both these constraints are intersected (and not or) which produces a convex set. Thus the problem is a convex optimization problem. Its concavity in this case means that a local maximum will also be the global maximum, becasue on a concave function there is only one maximum.

**(h)** The objective function is minimizing a convex function which makes it a convex optimization problem as long as the feasable region is also convex. In this case the feesable region is a sublevel set where $f(x) \leq a$ where if $f(x) is convex then the set is convex. Therfore since the objective function and the feesable reagion are convex the problem is a convex optimization problem. This convexity means thata local mainimum will also be the global minimum, becasue on a convex function there is only one minimum.  If you change the objective to maximize it is no longer a convex optimization problem. Likewise there is no gaurentee that a local maximum is a global maximum. This does not make the problem difficult to solve in this case though becase since there is only one conidtion the objective function is subject as well as since there is only two objective variables to this probmem could be solved fairly easily through graphically. Finally convexity alone cannot guarentee that there is an optimal soloution or that one exists becasue the set could be unbounded or have infinitely many solutions. 

## Computational setup

Run this supplied cell before working on Problems 2 and 3. No package installation is needed. Use `termination_status(model)` to check the solver result before reading `value(...)` or `objective_value(model)`. For these LPs, verify that the status is `MOI.OPTIMAL`.


In [1]:
using JuMP, HiGHS, Plots
import MathOptInterface as MOI


## Problem 2 — Spider-Man's chemical production

Spider-Man needs chemicals A, B, and C to make web fluid. He can run either of two production processes for any nonnegative number of hours; fractional hours are allowed. Outputs and costs are proportional to operating time.

| Process | Cost (dollars/hour) | A (vials/hour) | B (vials/hour) | C (vials/hour) |
| :-- | --: | --: | --: | --: |
| 1 | 4 | 3 | 1 | 1 |
| 2 | 1 | 1 | 1 | 0 |

He needs at least 10 vials of A, 5 vials of B, and 3 vials of C. Excess production is allowed and has no resale value or disposal cost. There is no upper bound on operating time.

### 2(a): Mathematical formulation

Formulate an LP that determines how long each process should run to meet the requirements at minimum cost. Define the decision variables and their units. State the objective, every constraint, and variable domains.


**Formulation for 2(a):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

**decision variables:** \
    $P_1$: Time process 1 is run (hours)  $[0, +\infty)$\
    $P_2$: Time process 2 is run (hours)  $[0,+\infty)$

**constraints:**\
    $P_2 \geq 0$ (non-0 constraint)\
    $3*P_1 + 1*P_2 \geq 10$ (vials A made)\
    $P_1 +P_2 \geq 5$ (vials B made)\
    $P_1 \geq 3$ (vials C made)

**Objective**\
    Minimize $4*P_1+1*P_2$ (cost in dollars)


### 2(b): Graphical solution

Before solving with JuMP, draw the constraint boundaries, label the feasible side of each, and shade their intersection. Include nonnegativity. Draw at least two parallel isocost lines and indicate the direction of decreasing cost. Identify the optimal point and cost from your graph and supporting algebra.

You may draw by hand and embed a picture, or use Julia/Plots. A finite plotting window is not an additional model constraint: indicate any direction in which the feasible region continues.


**Graphical solution and reasoning for 2(b):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.


 Optional: draw your graph using Plots here.
Alternatively, embed a handwritten graph in the Markdown answer above.

![Problem 2b graph.jpg](student-work/hw01/Problem_2b_graph.jpg)


### 2(c): Julia/JuMP solution and interpretation

Implement your formulation, solve it with HiGHS, and print the solver status, both process durations, and the total cost. Report how much of each chemical is produced and verify every requirement. Compare with your graphical solution.

Is the feasible region bounded? Does this LP nevertheless have a finite optimal cost? Explain. Identify a redundant chemical requirement, if one exists, and justify redundancy for all feasible points, not just at the optimum.


In [23]:
# Write your Julia/JuMP code here.

using JuMP, HiGHS
Spiderman=Model(HiGHS.Optimizer)
@variable(Spiderman,3<=P1) #Process 1 Time/vials C made
@variable(Spiderman, 0<=P2) #Process 2 Time
@constraint(Spiderman,10<=3P1+P2) #Vials A made
@constraint(Spiderman, 5 <= P1+P2) #Vials B made
@objective(Spiderman, Min, 4*P1+P2) #cost

optimize!(Spiderman)

status = termination_status(Spiderman)
status == MOI.OPTIMAL || error("HiGHS stopped with status $(status).")

process_1_time = value(P1)
process_2_time = value(P2)
cost_value = objective_value(Spiderman)

println("Termination status: ", status)
println("Produce ", process_1_time, " hours of Process 1.")
println("Produce ", process_2_time, " hours of Process 2.")
println("Spiderman Produced ", 3*process_1_time+process_2_time, " of vial A.")
println("Spiderman Produced ", process_1_time+process_2_time, " of vial B.")
println("Spiderman Produced ", process_1_time, " of vial C.")
println("Total cost: \$", cost_value)

Running HiGHS 1.15.1 (git hash: 04024d701f): Copyright (c) 2026 under MIT licence terms
Includes third-party software components, see THIRD_PARTY_NOTICES.md for full details
Using BLAS: blastrampoline 
LP has 2 rows; 2 cols; 4 nonzeros
Coefficient ranges:
  Matrix  [1e+00, 3e+00]
  Cost    [1e+00, 4e+00]
  Bound   [3e+00, 3e+00]
  RHS     [5e+00, 1e+01]
Presolving model
2 rows, 2 cols, 4 nonzeros 0s
0 rows, 0 cols, 0 nonzeros 0s
Presolve reductions: rows 0(-2); columns 0(-2); nonzeros 0(-4) - Reduced to empty
Performed postsolve
Solving the original LP from the solution after postsolve

Model status        : Optimal
Objective value     :  1.4000000000e+01
P-D objective error :  0.0000000000e+00
HiGHS run time      :          0.00
Termination status: OPTIMAL
Produce 3.0 hours of Process 1.
Produce 2.0 hours of Process 2.
Spiderman Produced 11.0 of vial A.
Spiderman Produced 5.0 of vial B.
Spiderman Produced 3.0 of vial C.
Total cost: $14.0


**Results and interpretation for 2(c):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

The results from Julia/JuMP agree with the graphical soloution where spiderman should produce 3 hours of Process 1 and 2 hrs of process 2 for a minimum cost of 14 dollars. In doing so he will produce 11 A vials, 5 B vials and, 3 C vials which is greater than or euqal to the 10 A, 5B and 3C vials he needed.  This optimization problem had a feasable region that was unbounded on on end but bounded on the other so if you were to try to maximize the function instead of minimize there would have been infinitely many soloutions becasue P1 and P2 had domains of $[0,+\infty)$ and the constraints only added constraints to the lower bound of the domain. Thus because we were trying to minimize the objective funciton we resulted in a bounded problem with a finite optimal cost. This problem also had a redundant constraing of $3*P_1 + 1*P_2 \geq 10$ (vials A made). This was redundant because it falls entirely outside of the feesable regions created by the constrints $P_2 \geq 0$ $P_1 +P_2 \geq 5$ and $P_1 \geq 3$. 

## Problem 3 — Planning a brewery's production

The 524 Brewing and Bottling Company produces light beer, pale ale, and malt liquor. It has 90 units of malt, 65 units of hops, and 80 units of yeast available. Resource requirements per gallon and profit per gallon are:

| Product | Malt (units/gallon) | Hops (units/gallon) | Yeast (units/gallon) | Profit (dollars/gallon) |
| :-- | --: | --: | --: | --: |
| Light beer | 2 | 3 | 2 | 2.50 |
| Pale ale | 2 | 2 | 1.5 | 2.00 |
| Malt liquor | 3 | 1.5 | 2 | 2.50 |

Any amount produced can be sold at the stated profit. Fractional gallons are allowed. There are no minimum production requirements initially.

### 3(a): Mathematical formulation

Formulate an LP to choose production quantities that maximize total profit. Define variables and units; state all resource constraints and variable domains.


**Formulation for 3(a):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

**decision variables:** \
    $LB$: Light Beer Produced (gallons)  $[0, +\infty)$\
    $PA$: Pale Ale Produced (gallons)  $[0, +\infty)$\
    $ML$: Malt Liquor Produced (gallons)  $[0,+\infty)$

**constraints:**\
    $2*LB+2*PA+3*ML \leq 90$ (Malt Constraint)\
    $3*LB+2*PA+1.5*ML \leq 65$ (Hops Constraint)\
    $2*LB+1.5*PA+2*ML \leq 80$ (Yeast Constraint)

**Objective**\
    Maximize $2.5*LB+2*PA+2.5*ML$ (Profit $)

### 3(b): Implement, solve, and interpret

Implement and solve your LP with HiGHS. Print the solver status, gallons of each product, total profit, and the amount of each resource used and left over.

Which products are produced, and which resources are fully used? Explain why having a positive profit per gallon does not ensure that a product will appear in an optimal plan. Relate your explanation to this instance rather than just stating a general rule.


In [7]:
# Write your Julia/JuMP code here.
using JuMP, HiGHS
Brew1=Model(HiGHS.Optimizer)
@variable(Brew1, 0<=LB) #Light Beer Produced (Gallons)
@variable(Brew1, 0<=PA) #Pale Ale Produced (Gallons)
@variable(Brew1, 0<=ML) #Malt Liquor Produced (Gallons)
@constraint(Brew1,2*LB+2*PA+3*ML<=90) #Malt Constraint
@constraint(Brew1,3*LB+2*PA+1.5*ML<=65) #Hops Constraint
@constraint(Brew1,2*LB+1.5*PA+2*ML<=80) #Yeast Constraint
@objective(Brew1, Max, 2.5*LB+2*PA+2.5*ML) #Profit
 
optimize!(Brew1)

status = termination_status(Brew1)
status == MOI.OPTIMAL || error("HiGHS stopped with status $(status).")

Light_Beer_Produced = value(LB)
Pale_Ale_Produced = value(PA)
Malt_Liquor_Produced = value(ML)
profit = objective_value(Brew1)

println("Termination status: ", status)
println("Produce ", Light_Beer_Produced, " gallons of Light Beer.")
println("Produce ", Pale_Ale_Produced, " gallons of Pale Ale.")
println("Produce ", Malt_Liquor_Produced, " gallons of Malt Liquor.")
println("This will use ", 2*Light_Beer_Produced+2*Pale_Ale_Produced+3*Malt_Liquor_Produced, " units of Malt")
println("This will have ", 90-(2*Light_Beer_Produced+2*Pale_Ale_Produced+3*Malt_Liquor_Produced), " units of Malt left")
println("This will use ", 3*Light_Beer_Produced+2*Pale_Ale_Produced+1.5*Malt_Liquor_Produced, " units of Hops")
println("This will have ", 65-(3*Light_Beer_Produced+2*Pale_Ale_Produced+1.5*Malt_Liquor_Produced), " units of Hops left")
println("This will use ", 2*Light_Beer_Produced+1.5*Pale_Ale_Produced+2*Malt_Liquor_Produced, " units of Yeast")
println("This will have ", 80-(2*Light_Beer_Produced+1.5*Pale_Ale_Produced+2*Malt_Liquor_Produced), " units of Yeast left")
println("Total profit: \$", profit)

Running HiGHS 1.15.1 (git hash: 04024d701f): Copyright (c) 2026 under MIT licence terms
Includes third-party software components, see THIRD_PARTY_NOTICES.md for full details
Using BLAS: blastrampoline 
LP has 3 rows; 3 cols; 9 nonzeros
Coefficient ranges:
  Matrix  [2e+00, 3e+00]
  Cost    [2e+00, 2e+00]
  Bound   [0e+00, 0e+00]
  RHS     [6e+01, 9e+01]
Presolving model
3 rows, 3 cols, 9 nonzeros 0s
3 rows, 3 cols, 9 nonzeros 0s
Presolve reductions: rows 3(-0); columns 3(-0); nonzeros 9(-0) - Not reduced
Problem not reduced by presolve: solving the LP
Using dual simplex solver
  Iteration        Objective     Infeasibilities num(sum)
          0    -6.9999926653e+00 Ph1: 3(19); Du: 3(6.99999) 0.0s
          2     8.3333333333e+01 Pr: 0(0) 0.0s

Model status        : Optimal
Simplex   iterations: 2
Objective value     :  8.3333333333e+01
P-D objective error :  8.4756588759e-17
HiGHS run time      :          0.00
Termination status: OPTIMAL
Produce 10.0 gallons of Light Beer.
Produce 0.0

**Results and interpretation for 3(b):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

In this optimization problem the brewer should produce 10 Gallons of light beer, 0 gallons of pale ale and 23.3 gallons of Malt Liquor inorder to make the maximum profit of $ 83.33.  This will use 90 units of Malt, 65 units of hops and 66.66 units of yeast, meaning all hops and malt will be used but there will be extra yeast left over. In this case pale ale does not show up in the soloution because even though it produces positive profit it produces 25% less profit while using only 15-20% less resources than the other options. 


### 3(c): Add a customer requirement

A customer now requires **at least 5 gallons of pale ale**. All other data remain unchanged.

Write the change to your mathematical formulation. Implement it and solve the revised LP. Report the new production quantities and total profit. How much profit is sacrificed to meet the requirement? Explain why adding a requirement cannot improve the maximum achievable profit when the objective and all other assumptions are unchanged.


**Formulation change and explanation for 3(c):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.


In [8]:
# Write your Julia/JuMP code here.
using JuMP, HiGHS
Brew1=Model(HiGHS.Optimizer)
@variable(Brew1, 0<=LB) #Light Beer Produced (Gallons)
@variable(Brew1, 5<=PA) #Pale Ale Produced (Gallons) Added 5 Gallon Minium Constraint
@variable(Brew1, 0<=ML) #Malt Liquor Produced (Gallons)
@constraint(Brew1,2*LB+2*PA+3*ML<=90) #Malt Constraint
@constraint(Brew1,3*LB+2*PA+1.5*ML<=65) #Hops Constraint
@constraint(Brew1,2*LB+1.5*PA+2*ML<=80) #Yeast Constraint
@objective(Brew1, Max, 2.5*LB+2*PA+2.5*ML) #Profit
 
optimize!(Brew1)

status = termination_status(Brew1)
status == MOI.OPTIMAL || error("HiGHS stopped with status $(status).")

Light_Beer_Produced = value(LB)
Pale_Ale_Produced = value(PA)
Malt_Liquor_Produced = value(ML)
profit = objective_value(Brew1)

println("Termination status: ", status)
println("Produce ", Light_Beer_Produced, " gallons of Light Beer.")
println("Produce ", Pale_Ale_Produced, " gallons of Pale Ale.")
println("Produce ", Malt_Liquor_Produced, " gallons of Malt Liquor.")
println("This will use ", 2*Light_Beer_Produced+2*Pale_Ale_Produced+3*Malt_Liquor_Produced, " units of Malt")
println("This will have ", 90-(2*Light_Beer_Produced+2*Pale_Ale_Produced+3*Malt_Liquor_Produced), " units of Malt left")
println("This will use ", 3*Light_Beer_Produced+2*Pale_Ale_Produced+1.5*Malt_Liquor_Produced, " units of Hops")
println("This will have ", 65-(3*Light_Beer_Produced+2*Pale_Ale_Produced+1.5*Malt_Liquor_Produced), " units of Hops left")
println("This will use ", 2*Light_Beer_Produced+1.5*Pale_Ale_Produced+2*Malt_Liquor_Produced, " units of Yeast")
println("This will have ", 80-(2*Light_Beer_Produced+1.5*Pale_Ale_Produced+2*Malt_Liquor_Produced), " units of Yeast left")
println("Total profit: \$", profit)

Running HiGHS 1.15.1 (git hash: 04024d701f): Copyright (c) 2026 under MIT licence terms
Includes third-party software components, see THIRD_PARTY_NOTICES.md for full details
Using BLAS: blastrampoline 
LP has 3 rows; 3 cols; 9 nonzeros
Coefficient ranges:
  Matrix  [2e+00, 3e+00]
  Cost    [2e+00, 2e+00]
  Bound   [5e+00, 5e+00]
  RHS     [6e+01, 9e+01]
Presolving model
3 rows, 3 cols, 9 nonzeros 0s
3 rows, 3 cols, 9 nonzeros 0s
Presolve reductions: rows 3(-0); columns 3(-0); nonzeros 9(-0) - Not reduced
Problem not reduced by presolve: solving the LP
Using dual simplex solver
  Iteration        Objective     Infeasibilities num(sum)
          0    -6.9999926653e+00 Ph1: 3(19); Du: 3(6.99999) 0.0s
          2     8.2916666667e+01 Pr: 0(0) 0.0s

Model status        : Optimal
Simplex   iterations: 2
Objective value     :  8.2916666667e+01
P-D objective error :  0.0000000000e+00
HiGHS run time      :          0.00
Termination status: OPTIMAL
Produce 7.5 gallons of Light Beer.
Produce 5.0 

**Results and interpretation for 3(c):**

Type your response (LaTeX is welcome), or insert a photograph/scan of your handwritten response here. Include the required reasoning.

In this optimization problem the brewer should produce 7.5 Gallons of light beer, 5 gallons of pale ale and 21.66 gallons of Malt Liquor inorder to make the maximum profit of 82.92 dollars. By adding the additional constraint of a minimum of 5 gallons of pale ale there will be a loss of profit of $ 0.41 This will use 90 units of Malt, 65 units of hops and 66.83 units of yeast, meaning all hops and malt will be used but there will be extra yeast left over. If you add a extra constraint to a problem you can never produce a greater outcome because you will either be adding a redundent constraint and thus wont change the result or you will add a constraint which reduces the size of the feesable region. This is becasue any constriant added inorder to maintain convexity of the problem must be intersected with the existing contraints and intersected sets results can never be bigger than either of the original sets.


## Collaboration and LLM-use statement — required

Complete this statement as the **very last part of your homework**. Be specific enough that a reader can understand what work you did yourself and how others or LLMs helped you learn.

- **People:** Give the name of every person you collaborated with, identify the problem/subpart you discussed, and explain the nature of the collaboration. If you worked alone, explicitly state that you did not collaborate with anyone.
- **LLMs/tools:** Name every LLM or AI assistant you used and identify the model/version as precisely as available. If the interface did not show a model/version, say so. If you used none, explicitly state that you did not use an LLM or AI assistant.
- **Specific assistance and learning:** For each tool, identify the problem/subpart, describe what you asked it to help with, what suggestions you used or rejected, and how the interaction helped you understand the mathematics or implementation. Explain how you checked the suggestions. “I used ChatGPT for help” is not sufficiently specific.

Disclosure does not replace the requirement that your submitted formulation, implementation, and explanation be your own work and that you understand them.


**People and collaboration:**

I worked alone on this homework assignment.

**LLMs/tools and models:**

I used Google Gemini 3.5 flash-lite model for assistance on this homework assignment.

**How I used the assistance, what I learned, and how I checked it:**

[Specific details for each tool and relevant problem/subpart; state not applicable if you used none.]

I spacificly used it to help remind me of all the meanings of the math notation in problem 1-f because I could not remember the exact meaning of the following structure $P=\{x\in\mathbb{R}^n:Ax\leq b\}$ and how it related to what it would mean if P was empty. From the discussion with this model I came to understand that an empty P would mean the writing remains the same it is just evaluated at an n dimentional 0 which I intuitively understood before the chat but needed assistance in properly notating it to make sure I explained it properly. 